internal/pkg/bootstrap/handlers/database/database.go

```diff
@@ -63,18 +63,23 @@ func NewDatabaseForCoreData(httpServer httpServer, database interfaces.Database)
 }
 
 // Return the dbClient interface
-func (d Database) newDBClient(loggingClient logger.LoggingClient) (dbInterfaces.DBClient, error) {
+func (d Database) newDBClient(loggingClient logger.LoggingClient, dic *di.Container) (dbInterfaces.DBClient, error) {
```

Comment:
Don't pass container around; pull the required dependencies out of it and pass those explicitly into this method as parameters.